profiler/src/main.rs

```diff
@@ -160,7 +160,15 @@ struct PerfResult {
 
 // Prepare file for benchmarking, possibly appending random data to it if needed.
 fn prepare_data_file(path: &Path, size: u64) -> eyre::Result<()> {
-    let file = OpenOptions::new().write(true).create(true).open(path)?;
+    if let Some(parent) = path.parent() {
+        std::fs::create_dir_all(parent)
+            .wrap_err_with(|| format!("creating directory {}", parent.display()))?;
+    }
+    let file = OpenOptions::new()
+        .write(true)
+        .create(true)
```

Comment:
GH actions complains that `.truncate(true)` should also be added to the options.